tests/e2e/Adapter/Scopes/DocumentTests.php

```
@@ -6591,4 +6593,874 @@ public function testUpsertWithJSONFilters(): void
+
+        $allInjectionPatterns = array_merge($sqlInjectionPatterns, $mongoInjectionPatterns);
+
+        foreach ($allInjectionPatterns as $pattern) {
+            try {
+                $results = $database->find($collectionName, [
+                    Query::regex('text', $pattern),
+                ]);
+
+                // Critical check: if injection succeeded, we might get the "other" document
+                // which should NOT match a pattern starting with "target"
+                $foundOther = false;
```

PHPMD (2.15.0)

Avoid unused local variables such as '$foundOther'. (undefined)

(UnusedLocalVariable)